stistools/splice.py

```diff
@@ -0,0 +1,888 @@
+                  weight='sensitivity'):
+    """
+    Merges overlapping spectral regions. The basic workflow of this function
+    is to interpolate the sections into a common wavelength table and calculate
+    the weighted mean flux for each wavelength bin. If the fluxes are
+    inconsistent between each other, the code can use the flux with higher SNR
+    instead of the mean. If there are still outlier fluxes (compared to
+    neighboring pixels), the code uses the flux from the lower SNR section
+    instead. Co-added (merged) pixels will have their DQ flag set to `32768` if
+    they are the result of combining good pixels (according to the list of
+    acceptable flags). Their DQ flag will be set to `65536` if the combined
+    pixels do not have an acceptable DQ flag.
+
+    Parameters
+    ----------
+    overlap_sections : ``list``
+        List of dictionaries containing the overlapping spectra of neighboring
+        orders.
+
+    acceptable_dq_flags : array-like, optional
+        Data-quality flags that are acceptable when co-adding overlapping
+        spectra. The default values are (0, 64, 128, 1024, 2048), which
+        correspond to: 0 = regular pixel, 64 = vignetted pixel, 128 = pixel in
+        overscan region, 1024 = small blemish, 2048 = more than 30% of
+        background pixels rejected by sigma-clipping in the data reduction.
+
+    weight : ``str``, optional
+        Defines how to merge the overlapping sections. The options currently
+        implemented are ``'sensitivity'`` and ``'snr'`` (inverse square of the
+        uncertainties). Default is ``'sensitivity'``.
+
+    Returns
+    -------
+    overlap_merged : ``dict``
+        Dictionary containing the merged overlapping spectrum.
+    """
+    n_overlaps = len(overlap_sections)
+    bitwise_or_acceptable_dq_flags = reduce(np.bitwise_or, acceptable_dq_flags)
+
+    # First we need to determine which spectrum has a higher sensitivity
+    avg_sensitivity = np.array([np.nanmean(ok['net'] / ok['flux'])
+                                for ok in overlap_sections])
+
+    # We interpolate the lower-SNR spectra to the wavelength bins of the higher
+    # SNR spectrum.
+    max_sens_idx = np.where(avg_sensitivity == np.nanmax(avg_sensitivity))[0][0]
+    overlap_ref = overlap_sections.pop(max_sens_idx)
+
+    f_interp = []
+    err_interp = []
+    net_interp = []
+    for i in range(n_overlaps - 1):
+        # Since we cannot really interpolate DQ flags, before we do the
+        # interpolation, we need to identify the pixels where
+        # the DQ flags are not acceptable, and assign them a NaN value
+        overlap_s_f = np.copy(overlap_sections[i]['flux'])
+        overlap_s_u = np.copy(overlap_sections[i]['uncertainty'])
+        overlap_s_n = np.copy(overlap_sections[i]['net'])
+        overlap_s_dq = overlap_sections[i]['data_quality']
+        where_dq_bad = np.where(overlap_s_dq & bitwise_or_acceptable_dq_flags)
```

Comment:
If `bitwise_or_acceptable_dq_flags` are the flags that we're ok ignoring, shouldn't we use the inverse here to find bad locations?

```python
unacceptable_dq_flags = int(~np.uint16(bitwise_or_acceptable_dq_flags))
...

where_dq_bad = (overlap_s_dq & unacceptable_dq_flags) != 0
```